nano/node/wallet.cpp

```diff
@@ -1471,11 +1471,13 @@ void nano::work_watcher::watching (nano::qualified_root const & root_a, std::sha
 											current->second = block;
 										}
 									}
-									watcher_l->node.network.flood_block (block, false);
-									assert (!watcher_l->node.active.update_difficulty (block));
-									watcher_l->update (root_a, block);
+									if (!watcher_l->node.active.update_difficulty (block))
```

Comment:
Does this need a test since it's a new control flow?

Reply:
Hard to test this since the block would have to be confirmed after a successful work generation but before the difficulty update. But I can perhaps change tests to ensure blocks are no longer under watch once confirmed

Reply:
Added tests to ensure blocks are removed when winning an election, but also on losing (this was not the case before, changed here).

Simplified this method and toned down the number of indentation levels.